src/state/medicals/reducer.ts

```diff
@@ -0,0 +1,38 @@
+import produce from "immer";
+import { IAction } from "../types";
+import {
+  GET_MEDICAL_FAIL,
+  GET_MEDICAL_LOADING,
+  GET_MEDICAL_SUCCESS,
+} from "./consts";
+import { initial } from "./initial";
+import { IMedicalState } from "./types";
+
+export default produce((draft: IMedicalState, action: IAction<any, any>) => {
+  switch (action.type) {
+    /**
+     * GET_MEDICALS
+     */
+    case GET_MEDICAL_LOADING: {
+      draft.medicalsOrderByName.status = "LOADING";
+      break;
+    }
+
+    case GET_MEDICAL_SUCCESS: {
+      if (action.payload.length > 0) {
+        draft.medicalsOrderByName.status = "SUCCESS";
+      } else {
+        draft.medicalsOrderByName.status = "SUCCESS_EMPTY";
+      }
```

Comment:
I don't understand the utility of this if statement: if the action dispatched it's SUCCESS, you should return the SUCCESS status too. The action should have a strict state handling. SUCCESS > SUCCESS , ERROR > ERROR and so on. 

The .length() method should be checked and handled inside the components.

Reply:
OK

Reply:
Yeah.

Remember to handler the length() control inside the App for both of them.

Reply:
if we have to remove the SUCCESS_EMPTY, we  will remove in all reducers. Is it so?

Reply:
In order to use that SUCCESS_EMPTY status we need a placeolder for empty table.